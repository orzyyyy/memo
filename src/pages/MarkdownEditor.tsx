import React from 'react';
import MarkDownDetail from './MarkdownDetail';
import { Col, Row, Input, Button } from 'antd';
const { TextArea } = Input;

export interface MarkdownEditorProps {
  onSave: () => void;
  value: string;
  onChange: (value: string) => void;
}
export interface MarkdownEditorState {
  textareaValue: string;
}

const MarkdownEditor = ({ value, onSave, onChange }: MarkdownEditorProps) => (
  <Row gutter={16} style={{ padding: 15, width: '100%' }}>
    <Col span={12}>
      <TextArea
        style={{ height: '90vh', fontSize: 16 }}
        onChange={e => onChange(e.target.value)}
        value={value}
      />
    </Col>
    <Col
      span={12}
      style={{
        height: '90vh',
        overflow: 'auto',
        paddingLeft: 24,
        fontSize: 16,
      }}
    >
      <MarkDownDetail dataSource={value} />
    </Col>
    <Col span={8} push={8} style={{ marginTop: 24 }}>
      <Button block onClick={onSave}>
        提交
      </Button>
    </Col>
  </Row>
);

export default MarkdownEditor;
