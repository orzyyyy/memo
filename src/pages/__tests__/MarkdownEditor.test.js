import React from 'react';
import { mount } from 'enzyme';
import MarkdownEditor from '../MarkdownEditor';
import 'nino-cli/scripts/setup';

describe('MarkdownEditor', () => {
  it('render correctly', () => {
    const wrapper = mount(
      <MarkdownEditor
        targetId="targetId"
        dataSource={'testDataSource'}
        onSave={null}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('onSave, onChange works', () => {
    const onSave = jest.fn();
    const wrapper = mount(
      <MarkdownEditor
        targetId="targetId"
        dataSource={'testDataSource'}
        onSave={onSave}
      />,
    );
    wrapper
      .find('Button')
      .props()
      .onClick();
    expect(onSave).toHaveBeenCalled();
    wrapper
      .find('TextArea')
      .props()
      .onChange({ target: { value: 'test' } });
    expect(wrapper.state().textareaValue).toBe('test');
  });
});
