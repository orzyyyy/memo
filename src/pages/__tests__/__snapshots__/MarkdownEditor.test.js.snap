// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownEditor render correctly 1`] = `
<MarkdownEditor
  dataSource="testDataSource"
  onSave={null}
  targetId="targetId"
>
  <Row
    gutter={16}
    style={
      Object {
        "padding": 15,
        "width": "100%",
      }
    }
  >
    <div
      className="ant-row"
      style={
        Object {
          "marginLeft": -8,
          "marginRight": -8,
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <Col
        span={12}
      >
        <div
          className="ant-col ant-col-12"
          style={
            Object {
              "paddingLeft": 8,
              "paddingRight": 8,
            }
          }
        >
          <TextArea
            onChange={[Function]}
            style={
              Object {
                "fontSize": 16,
                "height": "90vh",
              }
            }
          >
            <ResizeObserver
              disabled={true}
              onResize={[Function]}
            >
              <textarea
                className="ant-input"
                onChange={[Function]}
                onKeyDown={[Function]}
                style={
                  Object {
                    "fontSize": 16,
                    "height": "90vh",
                  }
                }
                value=""
              />
            </ResizeObserver>
          </TextArea>
        </div>
      </Col>
      <Col
        span={12}
        style={
          Object {
            "fontSize": 16,
            "height": "90vh",
            "overflow": "auto",
            "paddingLeft": 24,
          }
        }
      >
        <div
          className="ant-col ant-col-12"
          style={
            Object {
              "fontSize": 16,
              "height": "90vh",
              "overflow": "auto",
              "paddingLeft": 24,
              "paddingRight": 8,
            }
          }
        >
          <_default>
            <div
              className="MarkdownDetail markdown-body"
            >
              <div
                className="wrapper"
              >
                <div
                  className="content"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "",
                    }
                  }
                />
              </div>
            </div>
          </_default>
        </div>
      </Col>
      <Col
        push={8}
        span={8}
        style={
          Object {
            "marginTop": 24,
          }
        }
      >
        <div
          className="ant-col ant-col-8 ant-col-push-8"
          style={
            Object {
              "marginTop": 24,
              "paddingLeft": 8,
              "paddingRight": 8,
            }
          }
        >
          <Button
            block={true}
            ghost={false}
            htmlType="button"
            loading={false}
            onClick={null}
          >
            <Wave>
              <button
                className="ant-btn ant-btn-block"
                onClick={[Function]}
                type="button"
              >
                <span
                  key=".0"
                >
                  提 交
                </span>
              </button>
            </Wave>
          </Button>
        </div>
      </Col>
    </div>
  </Row>
</MarkdownEditor>
`;
